src/utils/useFilter.tsx

```diff
@@ -0,0 +1,39 @@
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith, SearchParams } from './searchHelper';
+
+export function useFilter() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || null;
+  const activeCenturies = searchParams.getAll('centuries') || [];
+
+  const handleSearch = (params: SearchParams) => {
+    setSearchParams(current => getSearchWith(current, params));
+  };
+
+  const getSortParams = (sortBy: string) => {
+    if (sortBy === sort && !order) {
+      return { sort: sortBy, order: 'desc' };
+    }
+
+    if (sortBy !== sort && !order) {
+      return { sort: sortBy, order: null };
```

Comment:
The condition `if (sortBy !== sort && !order)` sets the order to `null`. This might not be the intended behavior if you want to toggle between ascending and descending order. Consider revising this logic to handle sorting states more intuitively, such as toggling between 'asc', 'desc', and null.